newrelic-agent/src/main/java/com/newrelic/agent/jfr/JfrService.java

```diff
@@ -49,15 +52,15 @@ protected void doStart() {
 
                 final JFRUploader uploader = buildUploader(daemonConfig);
                 uploader.readyToSend(new EventConverter(commonAttrs));
-                final JfrController jfrController = SetupUtils.buildJfrController(daemonConfig, uploader);
+                jfrController = SetupUtils.buildJfrController(daemonConfig, uploader);
 
                 ExecutorService jfrMonitorService = Executors.newSingleThreadExecutor();
                 jfrMonitorService.submit(
                         new Runnable() {
                             @Override
                             public void run() {
                                 try {
-                                    jfrController.loop();
+                                    startJfrLoop();
```

Comment:
In tests, I wanted to verify the loop was started and no exception. By wrapping this in a local method of this class, I can spy and verify interactions.